### DISCUSSION TITLE
Сalculating stream size is incorrect

### DISCUSSION THREAD

**Discussion by Kovsonq** [ctongfei/progressbar#116] [Q&A] [answered]
If stream size is larger that unsigned int it returns wrong value.

int available = is.available();

And even If I know the size of my file, I cant set it by setInitialMax() - because it will be overrided by wrapper:

public static InputStream wrap(InputStream is, ProgressBarBuilder pbb) {
        long size = Util.getInputStreamSize(is);
        if (size != -1 && pbb.initialMaxIsSet())
            pbb.setInitialMax(size);
        return new ProgressBarWrappedInputStream(is, pbb.build());
    }

**Comment by ctongfei** [comment, marked as answer]
OK -- so you are saying that, if `setInitialMax` is invoked in the builder, do not use the `Util.getInputStreamSize` function?

**Reply by Kovsonq** [reply]
Yes

**Reply by Kovsonq** [reply]
> OK -- so you are saying that, if `setInitialMax` is invoked in the builder, do not use the `Util.getInputStreamSize` function?

So can I wait some changes for this wrapper?
Thanks

**Reply by ctongfei** [reply]
This will be in the next version, but I cannot guarantee when it will be released.

**Reply by Kovsonq** [reply]
Thanks a lot.
